docs/concepts/projects/config.md

```diff
@@ -19,20 +19,80 @@ affects selection of dependency versions (they must support the same Python vers
 
 ## Entry points
 
-Projects may define entry points for the project in the `[project.scripts]` table of the
-`pyproject.toml`.
+[Entry points](https://packaging.python.org/en/latest/specifications/entry-points/#entry-points) are
+the official term for an installed package to advertise interfaces. These include:
 
-For example, to declare a command called `hello` that invokes the `hello` function in the
-`example_package_app` module:
+- [Command line interfaces]()
+- [Graphical user interfaces]()
+- [Plugin entry points]()
+
+!!! important
+
+    Using the entry point tables requires a [build system](#build-systems) to be defined.
+
+### Command-line interfaces
+
+Projects may define command line interfaces (CLIs) for the project in the `[project.scripts]` table
+of the `pyproject.toml`.
+
+For example, to declare a command called `hello` that invokes the `hello` function in the `example`
+module:
 
 ```toml title="pyproject.toml"
 [project.scripts]
-hello = "example_package_app:hello"
+hello = "example:hello"
 ```
 
-!!! important
+Then, the command can be run from a console:
+
+```console
+$ uv run hello
+```
+
+### Graphical user interfaces
+
+Projects may define graphical user interfaces (GUIs) for the project in the `[project.gui-scripts]`
+table of the `pyproject.toml`.
+
+For example, to declare a command called `hello` that invokes the `app` function in the `example`
+module:
+
+```toml title="pyproject.toml"
+[project.gui-scripts]
+hello = "example:app"
+```
+
+These are only different from [command-line interfaces](#command-line-interfaces) on Windows, where
+they are wrapped by a GUI executable so they can be started without a console. On other platforms,
+they behave the same.
```

Comment:
I'd put this front in this section, we can even roll the entire section into a paragraph at the end of the previous paragraph:

> On windows, `[project.scripts]` will create a terminal. If you're writing a GUI application and don't want a terminal on Windows, use `[project.gui-scripts]` instead.